jina/types/arrays/search_ops.py

```diff
@@ -0,0 +1,69 @@
+import re
+import operator
+from typing import Dict
+
+import numpy as np
+
+
+class DocumentArraySearchOpsMixin:
+    """ A mixin that provides search functionality to DocumentArrays"""
+
+    operators = {
+        '<': operator.lt,
+        '>': operator.gt,
+        '==': operator.eq,
+        '!=': operator.ne,
+        '<=': operator.le,
+        '>=': operator.ge,
+    }
+
+    def find(
+        self,
+        regexes: Dict,
+        traversal_paths: list = ['r'],
+        operator: str = '>=',
+        value: int = 1,
+    ) -> None:
+        """
+        Find Documents that match the regular expressions in `regexes`.
```

Comment:
You might want to reword this to `Find Documents whose tags match` - on first reading it was not clear to me what these regexes are compared to.

Reply:
I totally agree